src/coreclr/debug/shared/dbgtransportsession.cpp

```diff
@@ -156,7 +156,7 @@ HRESULT DbgTransportSession::Init(DebuggerIPCControlBlock *pDCB, AppDomainEnumer
         Release();
         return E_OUTOFMEMORY;
     }
-
+    SetThreadName(m_hTransportThread, W(".NET Debug Transport"));
```

Comment:
Can this be moved into the `TransportWorkerStatic` method so that it is executed on the thread itself and thus works on more OSes?

Reply:
```
  Id   Target Id                                            Frame
* 1    Thread 0x7ffff7a4c3c0 (LWP 319371) "corerun"         __pthread_kill_implementation (no_tid=0, signo=6, threadid=140737348158400) at ./nptl/pthread_kill.c:44
  2    Thread 0x7ffff6afc640 (LWP 319374) "corerun-ust"     syscall () at ../sysdeps/unix/sysv/linux/x86_64/syscall.S:38
  3    Thread 0x7ffff62fb640 (LWP 319375) "corerun-ust"     syscall () at ../sysdeps/unix/sysv/linux/x86_64/syscall.S:38
  4    Thread 0x7ffff5afa640 (LWP 319376) ".NET SynchManag" 0x00007ffff7b65d7f in __GI___poll (fds=0x0, nfds=0, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
  5    Thread 0x7ffff52f9640 (LWP 319377) ".NET EventPipe"  0x00007ffff7b65d7f in __GI___poll (fds=0x7fff70008f20, nfds=1, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
  6    Thread 0x7ffff4ad3640 (LWP 319378) "corerun"         0x00007ffff7b61764 in __libc_open64 (file=0x55555559fa44 "/tmp/clr-debug-pipe-319371-10777213-in", oflag=0) at ../sysdeps/unix/sysv/linux/open64.c:41
  7    Thread 0x7fffe7fff640 (LWP 319379) ".NET Debugger"   __futex_abstimed_wait_common64 (private=0, cancel=true, abstime=0x0, op=393, expected=0, futex_word=0x555555644de0) at ./nptl/futex-internal.c:57
  8    Thread 0x7fff77cc4640 (LWP 319380) ".NET Finalizer"  __futex_abstimed_wait_common64 (private=1432319836, cancel=true, abstime=0x7fff77cc30b8, op=137, expected=0, futex_word=0x555555665950) at ./nptl/futex-internal.c:57
  9    Thread 0x7fffe6e02640 (LWP 319381) ".NET SigHandler" __GI___libc_read (nbytes=1, buf=0x7fffe6e01e47, fd=33) at ../sysdeps/unix/sysv/linux/read.c:26
```

When `SetThreadName` is moved to thread body, it doesn't change on Linux. (Tried on both `TransportWorkerStatic` and `TransportWorker`)

Reply:
Does that mean that you can either have the good thread names on macOS or Linux, but on not on both - unless you duplicate the thread name calls on both sides?

Reply:
This was my mistake and fixed to set name inside the thread function.